update decap docs (#6800)
withastro/docs: src/content/docs/en/guides/cms/decap-cms.mdx

---
import Grid from '~/components/FluidGrid.astro'
import Card from '~/components/ShowcaseCard.astro'

> import FileTree from "~/components/FileTree.astro"

Reviewer: ```suggestion
import { FileTree } from '@astrojs/starlight/components'
```
It's a Starlight component now! Need to update the contributing docs, too!

## Installing DecapCMS

There are two options for adding Decap to Astro:

1. [Install Decap via a package manager](https://decapcms.org/docs/install-decap-cms/#installing-with-npm) with the following command:

    <PackageManagerTabs>
      <Fragment slot="npm">
      ```shell
      npm install decap-cms-app
      ```
      </Fragment>
      <Fragment slot="pnpm">
      ```shell
      pnpm add decap-cms-app
      ```
      </Fragment>
      <Fragment slot="yarn">
      ```shell
      yarn add decap-cms-app
      ```
      </Fragment>
    </PackageManagerTabs>
 
 
1. Import the package into a `<script>` tag in your page `<body>`

	```html title='/admin'
    <body>
      <!-- Include the script that builds the page and powers Decap CMS -->
      <script src="https://unpkg.com/decap-cms@^3.1.2/dist/decap-cms.js"></script>
    </body>
	```
 

## Configuration

1. Create a static admin folder at `public/admin/`
2. Add `config.yml` to `public/admin/`: 
    <FileTree>
      - public
        - admin
          - config.yml
    </FileTree>

> 3. Configure the `collections` schema in `config.yml`. The following example configures a `blog` collection, defining a `label` for each entry's frontmatter property:

Reviewer: ```suggestion
3. To add support for content collections, configure each `collections` schema in `config.yml`. The following example configures a `blog` collection, defining a `label` for each entry's frontmatter property:
```
Originally, I had an "optional if using content collections" note here, but without that, just saying "the collection schema" didn't really make it clear what "the collection" was.

So, Hope it's OK to add this back in like this!

    ```yml title="/public/admin/config.yml"
    collections:
      - name: "blog" # Used in routes, e.g., /admin/collections/blog
        label: "Blog" # Used in the UI
        folder: "src/content/blog" # The path to the folder where the documents are stored
        create: true # Allow users to create new documents in this collection
        fields: # The fields for each document, usually in front matter
          - { label: "Layout", name: "layout", widget: "hidden", default: "blog" }
          - { label: "Title", name: "title", widget: "string" }
          - { label: "Publish Date", name: "date", widget: "datetime" }
          - { label: "Featured Image", name: "thumbnail", widget: "image" }
          - { label: "Rating (scale of 1-5)", name: "rating", widget: "number" }
          - { label: "Body", name: "body", widget: "markdown" }
    ```

4. Add the `admin` route for your React app. This file can be either `public/admin/index.html` alongside your `config.yml` or, if you prefer to use an Astro route, `src/pages/admin.astro`.

>     <FileTree>
>         - public
>           - admin
>             - config.yml
>             - index.html
>     </FileTree>

Reviewer: ```suggestion
    <FileTree>
     - public
       - admin
          - config.yml
          - index.html
    </FileTree>
```
FileTree probably doesn't want indentation to start with... let's see if this fixes it.

---
import Grid from '~/components/FluidGrid.astro'
import Card from '~/components/ShowcaseCard.astro'
import FileTree from "~/components/FileTree.astro"

[Decap CMS](https://www.decapcms.org/) (formerly Netlify CMS) is an open-source, Git-based content management system.

Decap allows you to take full advantage of all of Astro's features, including image optimization and content collections.

> Decap adds a route (typically `/admin`) to your project that will load a React app to allow authorized users to manage content directly from the deployed website. Decap will commit changes to your Astro project's source repository.

Reviewer: ```suggestion
Decap adds a route (typically `/admin`) to your project that will load a React app to allow authorized users to manage content directly from the deployed website. Decap will commit changes directly to your Astro project's source repository.
```

## Installing DecapCMS

There are two options for adding Decap to Astro:

> 1. [Install Decap via a package manager](https://decapcms.org/docs/install-decap-cms/#installing-with-npm) with the following command:

Reviewer: Non-blocking, but if pnpm and yarn are also acceptable, we could use the `<PackageManagerTabs/>` component here.

Author: done!

	```sh
	npm install decap-cms-app --save
	```
 
1. Import the package into a `<script>` tag in your page `<body>`

	```html title='/admin'
    <body>
      <!-- Include the script that builds the page and powers Decap CMS -->
      <script src="https://unpkg.com/decap-cms@^3.1.1/dist/decap-cms.js"></script>
    </body>
	```
 

> ## Configuration

Reviewer: Check the indentation of this section. All the code samples etc. need to be indented in further!

Author: done!

1. Create a static admin folder at `public/admin/`
2. Add `config.yml` to `public/admin/`: 
  <FileTree>

>   - admin
>     - index.html
>     - config.yml

Reviewer: ```suggestion
		- public
	    - admin
	    	- config.yml
```
let's show the public folder here!

  </FileTree>

> 3. Configure your collections schemas in `config.yml`

Reviewer: ```suggestion
3. (optional, if using content collections) Configure the `collections` schema in `config.yml`. The following example configures a `blog` collection, defining a `label` for each entry's frontmatter property:
```

  ```yml
  collections:
    - name: "blog" # Used in routes, e.g., /admin/collections/blog
      label: "Blog" # Used in the UI
      folder: "src/content/blog" # The path to the folder where the documents are stored
      create: true # Allow users to create new documents in this collection
      fields: # The fields for each document, usually in front matter
        - { label: "Layout", name: "layout", widget: "hidden", default: "blog" }
        - { label: "Title", name: "title", widget: "string" }
        - { label: "Publish Date", name: "date", widget: "datetime" }
        - { label: "Featured Image", name: "thumbnail", widget: "image" }
        - { label: "Rating (scale of 1-5)", name: "rating", widget: "number" }
        - { label: "Body", name: "body", widget: "markdown" }
  ```

> 5. (optional) To enable media uploads to a specific folder via the Decap editor, add an appropriate path:

Reviewer: ```suggestion

4. Add the `admin` route for your React app. This file can be either `public/admin/index.html` alongside your `config.yml` or, if you prefer to use an Astro route, `src/pages/admin.astro`.

		<FileTree>
				- public
			    - admin
			    	- config.yml
			    	- index.html
		</FileTree>

5. (optional) To enable media uploads to a specific folder via the Decap editor, add an appropriate path:
```
Is there a sample of what this file (html or astro) should look like? Can it just be a blank file, or do you need to put something in here? I would either show an example (if it can't be blank) or link to DecapCMS docs to an example of what this file looks like.

  ```yml
  media_folder: "src/assets/images" # Location where files will be stored in the repo 
  public_folder: "src/assets/images" # The src attribute for uploaded media
  ```

See the [the Decap CMS configuration documentation](https://decapcms.org/docs/configure-decap-cms/) for full instructions and options.

## Usage
Navigate to `yoursite.com/admin/` to use the Decap CMS editor.

> 
> ### Common practices
> 
> Many users will add their admin route and config file into their project's `public` folder. However it is equally viable to have this route be a `.astro` file in the [Pages](/en/basics/astro-pages/) directory.

Reviewer: ```suggestion
```
I still don't like this section.  :smile: I incorporated it above!

## Authentication

### Decap CMS with Netlify Identity

> Decap CMS was originally developed by Netlify and has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). When deploying to netlify, configure Identity for your project via the Netlify dashboard and include the [Netlify Identity Widget](https://github.com/netlify/netlify-identity-widget) on the `/admin` route of your project. Optionally include the Identity Widget on the homepage of your site if you plan to invite new users via email.

Reviewer: ```suggestion
Decap CMS was originally developed by Netlify and has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/).

When deploying to netlify, configure Identity for your project via the Netlify dashboard and include the [Netlify Identity Widget](https://github.com/netlify/netlify-identity-widget) on the `admin` route of your project. Optionally include the Identity Widget on the homepage of your site if you plan to invite new users via email.
```

import Grid from '~/components/FluidGrid.astro'
import Card from '~/components/ShowcaseCard.astro'

> [Decap CMS](https://www.decapcms.org/) (formerly Netlify CMS) is an open-source, Git-based content management system. When using Decap you will be able to use Astro's [Content Collections](/en/guides/content-collections/) as well as [Image Optimization](/en/guides/images/).

Author: This is very unique to git based CMSes, and specifically to Decap. I see what you mean though 🤔

For example Tina makes it hard to use Astro:assets Image Optimization

Basically keystatic, static (fork of decap) and decap are the only CMS that let you use both CC and astro:assets

> [Decap CMS](https://www.decapcms.org/) (formerly Netlify CMS) is an open-source, Git-based content management system. When using Decap you will be able to use Astro's [Content Collections](/en/guides/content-collections/) as well as [Image Optimization](/en/guides/images/).

Reviewer: Gotcha! Makes sense! Is Tina the only exception though? I often think of Keystatic, Decap and Tina as the *only* real Git-based options. So it *might* be that this is more a weakness of Tina than a property of Decap?

> Decap adds a route (typically `/admin`) to your project that will load a React app which allows authorized users to manage their content directly from the deployed website. Whenever a user makes changes to their content Decap will process those changes through git, which typically will trigger continuous deployment.

Reviewer 2: ```suggestion
Decap adds a route (typically `/admin`) to your project that will load a React app which allows authorized users to manage their content directly from the deployed website. Whenever a user makes changes to their content, Decap will process those changes through git, which typically will trigger continuous deployment.
```

## Integrating with Astro

There are two options for using Decap with Astro:

1. Loading the default package, either via [unpkg](https://www.unpkg.com) or importing the package into a `<script>` tag.

> 2. Manually initializing Decap which is documented by Decap [Decap Docs](https://decapcms.org/docs/manual-initialization/).

Reviewer 2: ```suggestion
2. [Manually initializing Decap](https://decapcms.org/docs/manual-initialization/).
```

### Common practices

> Many users will add their admin route and config file into their project's public folder. However it is equally viable to have this route be a `.astro` file in the Pages directory.

Reviewer 2: ```suggestion
Many users will add their admin route and config file into their project's `public` folder. However it is equally viable to have this route be a `.astro` file in the [Pages](/en/basics/astro-pages/) directory.
```

Reviewer 2: Or maybe just: `src/pages` directory

> Decap CMS was originally developed by Netlify so it has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). When deploying to hosting providers other than Netlify, it is necessary to create your own OAuth routes. These routes are often created as standalone serverless functions for static sites, but it is possible to use Astro's On-demand Rendering to create these routes within your Astro Project.  See [Decap's OAuth Docs](https://decapcms.org/docs/external-oauth-clients/) for more details.

Author: This is actually one of my main reasons for wanting to update this doc. Netlify has abandoned this project all together and it looks like their support for netlify identity is dropping in quality (saw complaint in decap discord)

Decap is not at all tied to netlify and I want to make that more clear

> Decap CMS was originally developed by Netlify so it has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). When deploying to hosting providers other than Netlify, it is necessary to create your own OAuth routes. These routes are often created as standalone serverless functions for static sites, but it is possible to use Astro's On-demand Rendering to create these routes within your Astro Project.  See [Decap's OAuth Docs](https://decapcms.org/docs/external-oauth-clients/) for more details.

Reviewer: OK, so then it's NOT a happy path! Gotcha!

In that case, instead of a "Common Practices" thing, then how about sections that literally are for those things.

```suggestion
## Adding Authentication

### Decap CMS with Netlify Identity

Decap CMS was originally developed by Netlify and has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). This allows you to have JSON web token (JWT)-based user authentication for your sites deployed to Netlify. You can use this for gated content, site administration, and more.

### Decap CMS with External OAuth Clients

When deploying to hosting providers other than Netlify, you must create your own OAuth routes.

In Astro, this can be done with on-demand rendered routes in your project configured with [`server` or `hybrid` output](/en/guides/server-side-rendering/#enable-on-demand-server-rendering) enabled.

See [Decap's OAuth Docs](https://decapcms.org/docs/external-oauth-clients/) for a list of compatible community-maintained OAuth clients.
```

> Decap CMS was originally developed by Netlify so it has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). When deploying to hosting providers other than Netlify, it is necessary to create your own Oauth routes. These routes are often created as standalone serverless functions for static sites, but it is possible to use Astro's On Demand Rendering to create these routes within your Astro Project.  See [Decap's Oauth Docs](https://decapcms.org/docs/external-oauth-clients/) for more details.

Reviewer 2: ```suggestion
Decap CMS was originally developed by Netlify so it has first class support for [Netlify Identity](https://docs.netlify.com/security/secure-access-to-sites/identity/). When deploying to hosting providers other than Netlify, it is necessary to create your own OAuth routes. These routes are often created as standalone serverless functions for static sites, but it is possible to use Astro's On-demand Rendering to create these routes within your Astro Project.  See [Decap's OAuth Docs](https://decapcms.org/docs/external-oauth-clients/) for more details.
```

Reviewer 2: Docs uses `on-demand rendering`
